Automattic/jetpack
# Forms: add post to url hook

This PR implements the post-to-url feature on the Forms package as a replacement to the (left behind) [Salesforce integration implementation on the plugin](https://github.com/Automattic/jetpack/pull/26903).

## Proposed changes:
Add a new class to hook on feedback post creation. If the form attributes specify a post-to-url, it will trigger an HTTP post using the submit values. Salesforce configurations are accepted and turned into a generic configuration as a backwards compatibility feature.

The change also adds new expected attributes on the form to pave the road for new features:
- hiddenFields: TBD, simple name/value pairs added to forms
- postToUrl: TBD, post to url configuration
- salesforceData (backwards compatibility with plugin)

### Other information:


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

Make sure you enable the package:
```php
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
```

Run install on both the plugin and package (jetpack install packages/forms && jetpack install plugins/jetpack) to ensure autoloader update.

Publish a post with a Salesforce Lead form (or use an existing one). Make sure you use a valid `organizationID` on the form setup. Fill the form and submit. You should see a new lead created on your Salesforce instance.

Salesforce instance can be acquired free for testing, I can't share mine. Use the form's instructions to get your organization ID.
